src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-details/rgw-bucket-details.component.ts

```diff
@@ -69,4 +69,9 @@ export class RgwBucketDetailsComponent implements OnChanges {
     });
     return data;
   }
+
+  switchLifecycleFormat(): void {
+    this.lifecycleFormat = this.lifecycleFormat === 'json' ? 'xml' : 'json';
```

Comment:
this is actually confusing.. i see what you are trying to do but it took me a while to actually digest that. why not just simplify it and get the format as input and then assign it to the lifecycleFormat?

```suggestion
  switchLifecycleFormat(format: string): void {
    this.lifecycleFormat = format;
```

Reply:
I wanted to have a toggle like behaviour but updated it to: `updateLifecycleFormatTo(format: 'json' | 'xml')` to favour readability